presto-main/src/main/java/io/prestosql/operator/ExchangeOperator.java

```diff
@@ -185,7 +185,7 @@ public Page getOutput()
         Page deserializedPage = serde.deserialize(page);
         operatorContext.recordProcessedInput(deserializedPage.getSizeInBytes(), page.getPositionCount());
 
-        return deserializedPage;
+        return deserializedPage.getLazyWrappedPage();
```

Comment:
this doesn't give any performance advantage. We would want to have lazy deserialization, see: `io.prestosql.execution.buffer.PagesSerde#deserialize`

Reply:
This is was just a wrap like we do for tpch connector. But we can't achieve lazy deserialization without having an index for each block in the serialized Page, if we can pass it as a part of serialized page then it might be possible

Reply:
I think that `io.prestosql.spi.block.BlockEncodingSerde#readBlock` should return `LazyBlock` instead